modules/core/src/lib/transform/transform.js

```diff
@@ -0,0 +1,166 @@
+    for (const resourceTransform of resourceTransforms) {
+      swapped = swapped || resourceTransform.swap();
+    }
+    assert(swapped, 'Nothing to swap');
+  }
+
+  // Return Buffer object for given varying name.
+  getBuffer(varyingName = null) {
+    return this.bufferTransform && this.bufferTransform.getBuffer(varyingName);
+  }
+
+  // Return data either from Buffer or from Texture
+  getData(opts = {}) {
+    const resourceTransforms = [this.bufferTransform, this.textureTransform].filter(Boolean);
+    for (const resourceTransform of resourceTransforms) {
+      const data = resourceTransform.getData(opts);
+      if (data) {
+        return data;
+      }
+    }
+    return null;
+  }
+
+  // Return framebuffer object if rendering to textures
+  getFramebuffer() {
+    return this.textureTransform && this.textureTransform.getFramebuffer();
+  }
+
+  // Update some or all buffer/texture bindings.
+  update(opts = {}) {
+    if (opts.elementCount) {
+      this.model.setVertexCount(opts.elementCount);
+    }
+    const resourceTransforms = [this.bufferTransform, this.textureTransform].filter(Boolean);
+    for (const resourceTransform of resourceTransforms) {
+      resourceTransform.update(opts);
+    }
+  }
+
+  // Private
+
+  initialize(props = {}) {
+    const {gl} = this;
+    this.buildResourceTransforms(gl, props);
+
+    props = this.getModelProps(props);
+    this.model = new Model(
+      gl,
+      Object.assign({}, props, {
+        fs: props.fs || getPassthroughFS({version: getShaderVersion(props.vs)}),
+        id: props.id || 'transform-model',
+        drawMode: props.drawMode || GL.POINTS,
+        vertexCount: props.elementCount
+      })
+    );
+
+    /* eslint-disable no-unused-expressions */
+    this.bufferTransform && this.bufferTransform.setupResources({model: this.model});
+    /* eslint-enable no-unused-expressions */
+  }
+
+  getModelProps(props) {
+    let updatedProps = Object.assign({}, props);
+    const resourceTransforms = [this.bufferTransform, this.textureTransform].filter(Boolean);
+    for (const resourceTransform of resourceTransforms) {
+      updatedProps = resourceTransform.getModelProps(updatedProps);
+    }
+    return updatedProps;
+  }
+
+  buildResourceTransforms(gl, props) {
+    if (this.canCreateBufferTransform(props)) {
+      this.bufferTransform = new BufferTransform(gl, props);
+    }
+    if (this.canCreateTextureTransform(props)) {
+      this.textureTransform = new TextureTransform(gl, props);
+    }
+    assert(
+      this.bufferTransform || this.textureTransform,
+      'must provide source/feedback buffers or source/target textures'
+    );
+  }
+
+  canCreateBufferTransform(props) {
+    if (!isObjectEmpty(props.sourceBuffers)) {
+      return true;
+    }
+    if (!isObjectEmpty(props.feedbackBuffers)) {
+      return true;
+    }
+    return false;
+  }
+
+  canCreateTextureTransform(props) {
```

Comment:
These two can be plain functions instead of class methods - method names cannot be minified and inflate bundle size.